### PR TITLE
feat: Add load-balance step extension

### PR DESCRIPTION
**Pull request by mmelko**
Adds Load-balance step extension: 

to test use this view-definition:
```
name: Config
id: step-extension-load-balance-eip
type: step
url: http://localhost:9001/remoteEntry.js
scope: 'loadBalanceStep'
module: './LoadBalanceStep'
constraints:
   -
      mandatory: true
      operation: CONTAINS_STEP_NAME
      parameter: load-balance
``` 
      
Screenshot 2023-02-23 at 14 11 37

### CODE REVIEW

**Comment by kahboom** [comment]
@mmelko how do you feel about putting some of that text inside of a popover? not required but I think it could look a bit neater.

File: load-balance/src/App.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import '@patternfly/patternfly/utilities/Flex/flex.css';
+import '@patternfly/patternfly/utilities/Sizing/sizing.css';
+import '@patternfly/patternfly/utilities/Spacing/spacing.css';
+import { LoadBalanceStep } from './LoadBalanceStep';
+
+function App() {
+  return (
+    <div className="App">
+      <header className="App-header">
+        <p>
```
Comment (lordrip) [inline load-balance/src/App.tsx:14]: [:nail_care:] Not a big deal, but could be possible to remove the unneeded tags? :)

File: load-balance/src/LoadBalanceStep.test.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    />);
+
+    const loadBalanceSelect = screen.getByTestId('load-balance-select');
+    expect(loadBalanceSelect).toBeInTheDocument();
+    //
+    const applyBtn = screen.getByTestId('lb-apply-button');
+    expect(applyBtn).toBeInTheDocument();
+    fireEvent.click(applyBtn);
+
+    console.log(step);
```
Comment (lordrip) [inline load-balance/src/LoadBalanceStep.test.tsx:26]: What are your thoughts about logging the step during the test? I fear that at some point we might add noise to the CI

```suggestion
```

File: load-balance/src/LoadBalanceStep.tsx [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+                setLbConfiguration(found.configuration);
+                setCurrentLoadBalancer(found);
+            }
+        }
+
+        const updateParameterValue = (key: string, newValue: string | number | boolean) => {
+            const tmpField: any = {...lbConfiguration.get(key), value: newValue};
+            const tmpProperties: Map<string, ILBConfigurationField> = lbConfiguration
+            tmpProperties.set(key, tmpField);
+            setLbConfiguration(new Map(tmpProperties));
```
Comment (lordrip) [inline load-balance/src/LoadBalanceStep.tsx:130]: Is it needed to create a new `Map` in every method call?

File: load-balance/src/LoadBalancerModel.tsx [hunk 4]
```diff
@@ -0,0 +1,81 @@
+export type ILBConfigurationField = {
```
Comment (lordrip) [inline load-balance/src/LoadBalancerModel.tsx:1]: This is cool :+1:

**Review by lordrip — APPROVED** [review]
LGTM, If possible could we remove the unneeded files? :)

File: load-balance/src/LoadBalanceStep.test.tsx (hunk 2 above)
Comment (mmelko) [reply load-balance/src/LoadBalanceStep.test.tsx:26]: I will delete it.

**Comment by mmelko** [comment]
@kahboom that would be good but I'd need to bring some logic there, I'm building that settings automatically from a model. But I guess we can improve this in the future.

File: load-balance/src/LoadBalanceStep.tsx (hunk 3 above)
Comment (mmelko) [reply load-balance/src/LoadBalanceStep.tsx:130]: technically yes, afaik reference would stay the same and the fields wouldn't re-render with new value.

**Review by lordrip — APPROVED** [review]

**Pull request merged**